zerver/webhooks/papertrail/view.py

```diff
@@ -17,12 +18,12 @@
 ```
 """.strip()
 
-def ensure_keys(name: str, data: Any) -> Optional[str]:
+def ensure_keys(name: str, data: Any) -> Any:
     if 'events' in data and 'saved_search' in data:
         saved_search = data['saved_search']
         if 'name' in saved_search and 'html_search_url' in saved_search:
-            return None
-    return _("Missing expected keys")
+            return data
+    raise ValidationError(_("Missing expected keys"))
```

Comment:
Should this be replaced by `check_dict`?  Seems like it doesn't do anything special.

Reply:
Yeah probably. Replaced. The validation could probably be further tightened by someone who knows what’s going on here or wants to figure it out.